### PR TITLE
Reset cross-signing before backup when resetting both

### PR DESCRIPTION
**Pull request by uhoreg**
so that the backup gets signed by the new cross-signing key

## Checklist

-   [ ] Tests written for new code (and old code if feasible).
-   [ ] New or updated `public`/`exported` symbols have accurate [TSDoc](https://tsdoc.org/) documentation.
-   [ ] Linter and other CI checks pass.
-   [ ] I have licensed the changes to Element by completing the [Contributor License Agreement (CLA)](https://cla-assistant.io/element-hq/element-web)

### CODE REVIEW

**Comment by uhoreg** [comment]
The Jest test looks like just a flaky test.  No idea why SonarCloud isn't reporting anything.  And earlier version of this PR had 70% test coverage.  I've since added another test, but I don't know if that's enough for SonarCloud until SonarCloud returns its result.  I think that this PR is otherwise ready for review.

**Review by t3chguy — APPROVED** [review]

**Comment by t3chguy** [comment]
If a jest test fails/flakes then Sonar won't run as it won't have a complete junit report to work from

File: src/async-components/views/dialogs/security/CreateSecretStorageDialog.tsx [hunk 1]
```diff
@@ -292,12 +294,28 @@ export default class CreateSecretStorageDialog extends React.PureComponent<IProp
...
+                /* Resetting cross-signing requires secret storage to be reset
+                 * (otherwise it will try to store the cross-signing keys in the
+                 * old secret storage, and may prompt for the old key, which is
+                 * probably not available), and resetting key backup requires
+                 * cross-signing to be reset (so that the new backup can be
+                 * signed by the new cross-signing key).  So we reset secret
+                 * storage first, then cross-signing, then key backup.
+                 */
                 logger.log("Forcing secret storage reset");
                 await crypto.bootstrapSecretStorage({
```
Comment (BillCarsonFr) [inline src/async-components/views/dialogs/security/CreateSecretStorageDialog.tsx:306]: Why not add the `resetCrossSigning` as an option to `bootstrapSecretStorage`?

**Review by BillCarsonFr — APPROVED** [review]
Just a question, otherwise ok

File: src/async-components/views/dialogs/security/CreateSecretStorageDialog.tsx (hunk 1 above)
Comment (uhoreg) [reply src/async-components/views/dialogs/security/CreateSecretStorageDialog.tsx:306]: Mainly just to avoid making such an invasive change.

**Pull request merged**